### PR TITLE
fix some math

### PR DESCRIPTION
**Pull request by mkstoyanov**
RMCE does not require 2 allocations and 3 loops of solution-size data.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -281,14 +281,14 @@ int main(int argc, char **argv)
...
-                          squared.size();
-        return std::sqrt(mean);
+      auto const RMSE = [&]() {
+        prec s{0}, d{0};
+        for (int i = 0; i < f_val.size(); i++)
+        {
+          d = f_val[i] - analytic_solution[i];
+          s += d * d;
+        }
+        return std::sqrt(s);
```
Comment (quantumsteve) [inline src/main.cpp]: The old code divides by `squared.size()` before taking the square root.

```suggestion
        prec s{0};
        for (int i = 0; i < f_val.size(); i++)
        {
          prec const d = f_val[i] - analytic_solution[i];
          s += d * d;
        }
        return std::sqrt(s / f_val.size());
```

**Review by quantumsteve — COMMENTED** [review]
Do we want RMSE of the vector or per element?

File: src/main.cpp (hunk 1 above)
Comment (mkstoyanov) [reply src/main.cpp]: Duh!

**Comment by mkstoyanov** [comment]
> Do we want RMSE of the vector or per element?

"mean" error means average across all elements, per-element will have to average over time or some external dimension, or it is simply an error vector of some sort. The old code did compute it down to a scalar, even if it is not obvious due to the overuse of `auto`

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**